src/test/java/redis/clients/jedis/tests/JedisSentinelPoolTest.java

```diff
@@ -161,6 +163,36 @@ public void customClientName() {
 
     assertTrue(pool.isClosed());
   }
+  
+  @Test(expected = JedisConnectionException.class)
+  public void testResetInvalidPassword() {
+    JedisFactory factory = new JedisFactory(null, 0, 2000, 2000, "foobared", 0, "my_shiny_client_name", false, null, null, null);
+    JedisSentinelPool pool = new JedisSentinelPool(MASTER_NAME, sentinels, new JedisPoolConfig(), factory);
+    Jedis obj1 = pool.getResource();
+    obj1.set("foo", "bar");
+    assertEquals("bar", obj1.get("foo"));
+    assertEquals(1, pool.getNumActive());
+    
+    factory.setPassword("wrong password");
```

Comment:
Let's wrap with try-catch and remove expected statement. The test will pass even JedisConnectionException is thrown from obj1.

Reply:
done

Reply:
done